atest/testdata/standard_libraries/builtin/wait_until_keyword_succeeds.robot

```diff
@@ -104,6 +104,28 @@ Keyword Not Found Inside Wait Until Keyword Succeeds
     [Documentation]    FAIL No keyword with name 'Non Existing KW' found.
     Wait Until Keyword Succeeds    1 second    0.1s    Non Existing KW
 
+Undefined Scalar Variable Inside Wait Until Keyword Succeeds
+    [Documentation]    FAIL STARTS: Variable '${var_nonex}' not found.
+    Wait Until Keyword Succeeds    1 second    0.1s    ${var_nonex}
+
+Undefined List Variable Inside Wait Until Keyword Succeeds
+    [Documentation]    FAIL STARTS: Variable '@{var_nonex}' not found.
+    Wait Until Keyword Succeeds    1 second    0.1s    @{var_nonex}
+
+Undefined Dictionary Variable Inside Wait Until Keyword Succeeds
+    [Documentation]    FAIL STARTS: Variable '&{var_nonex}' not found.
+    Wait Until Keyword Succeeds    1 second    0.1s    &{var_nonex}
+
+Extended Variable Syntax Using List Inside Wait Until Keyword Succeeds
+    [Documentation]    FAIL Keyword 'log' failed after retrying for 1 second. The last error was: Resolving variable '${list[2]}' failed: IndexError: list index out of range
+    ${list} =    Create List    1    2
+    wait until keyword succeeds    1 second    0.1s    log  ${list[2]}
+
+Extended Variable Syntax Using Dictionary Inside Wait Until Keyword Succeeds
+    [Documentation]    FAIL Keyword 'log' failed after retrying for 1 second. The last error was: Resolving variable '${dict['key_nonex']}' failed: KeyError: 'key_nonex'
+    ${dict} =  Create Dictionary  key1=val1  key2=val2
+    wait until keyword succeeds    1 second    0.1s    log  ${dict['key_nonex']}
+
```

Comment:
1. These tests have the same problem as `Run Keyword And Continue On Failure` tests above that the error would be same regardless does the exception exit WUKS or not.

2. I'm actually fairly certain that the three first test fail immediately. You are using non-existing variable as a keyword and I don't think WUKS will run any rounds in that case.

3. It would probably be easiest to verify that WUKS is actually executed several times in test execution side. Let me know if you need help with that. You could also try constructing WUKS where the variable first doesn't exist in the beginning but will be available later. Even in that case it would be a good idea to verify that WUKS has been executed more than once on the execution side.

4. Please use shorter timeout than 1s if it is going to be exhausted. It unnecessarily extends suite execution time.